src/device/generate.py

```diff
@@ -612,14 +640,22 @@ def partition_by_name(fns):
       (coll, algo, proto, redop, ty, acc, pipeline, unroll) = fn
       sym = paste("_", coll, algo, proto, redop, ty, acc, pipeline, unroll)
       if proto == "LL128":
-        out("#if (defined(__gfx90a__) || defined(__gfx942__) || defined(__gfx950__)) && defined(ENABLE_LL128)\n")
+        if acc == "0":
+          out("#if (defined(__gfx90a__) || defined(__gfx942__) || defined(__gfx950__)) && defined(ENABLE_LL128)\n")
+        else:
+          out("#if (defined(__gfx942__) || defined(__gfx950__)) && defined(ENABLE_LL128)\n")
+      else:
+        if acc == "1":
+          out("#if (defined(__gfx90a__) || defined(__gfx942__) || defined(__gfx950__))\n")
       out(
         "DEFINE_ncclDevFunc({sym}, ncclFunc{coll}, {redop_cxx}, {ty_cxx}, NCCL_ALGO_{algo}, NCCL_PROTO_{proto}, {acc}, {pipeline}, {unroll})\n"
         .format(sym=sym, coll=coll, redop_cxx=redop_to_cxx[redop], ty_cxx=ty_to_cxx[ty],
                 algo=(algo or "RING"), proto=(proto or "SIMPLE"), acc=acc, pipeline=pipeline, unroll=unroll)
       )
       if proto == "LL128":
         out("#endif\n")
+      if acc == "1" and proto != "LL128":
+        out("#endif\n")
```

Comment:
This should work, no?
```suggestion
      if proto == "LL128" or acc == "1":
        out("#endif\n")
```